zerver/views/realm.py

```diff
@@ -357,20 +365,35 @@ def update_realm(
 
         assert setting_group_id_name in req_group_setting_vars
 
-        if req_group_setting_vars[setting_group_id_name] is not None and req_group_setting_vars[
-            setting_group_id_name
-        ] != getattr(realm, setting_group_id_name):
-            user_group_id = req_group_setting_vars[setting_group_id_name]
+        setting_value = req_group_setting_vars[setting_group_id_name]
+
+        if setting_value is None:
+            continue
+
+        new_setting = getattr(setting_value, "new", setting_value)
+        new_setting_value = parse_group_setting_value(new_setting)
+
+        expected_current_setting_value = None
+        old_setting = getattr(setting_value, "old", None)
+        if old_setting is not None:
+            expected_current_setting_value = parse_group_setting_value(old_setting)
+
+        current_value = getattr(realm, setting_group_id_name)
+        if validate_group_setting_value_change(
+            current_value, new_setting_value, expected_current_setting_value
+        ):
+            setting_user_group = new_setting_value
```

Comment:
I had to make bunch of changes here. Do you expect them to be in a separate prep commit?